### PR TITLE
Msbuild warning fix for jenkins

### PR DESCRIPTION
**Pull request by amznestebanpapp**
Jenkins does not define a tmp directory, we were defining it, but it didnt get propagated to MSBuild.
It also didnt get picked up by wix. 

I added some messages to understand what was going on... on Jenkins:
```
[2021-11-22T17:42:55.321Z]   IntDirFullPath=D:\workspace\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.dir\profile\, OutDirFullPath=D:\workspace\o3de\build\windows_vs2019\lib\profile\, Tmp=, Temp=
```
The check comes from `Microsoft.CppBuild.targets`:
```
    <VCMessage Code="MSB8029" Type="Warning" Condition="'$(DesignTimeBuild)'!='true' and '$(IgnoreWarnIntDirInTempDetected)' != 'true' and ('$(IntDirFullPath.StartsWith($(Tmp), true, null))' == 'true' or '$(IntDirFullPath.StartsWith($(Temp), true, null))' == 'true' or '$(OutDirFullPath.StartsWith($(Tmp), true, null))' == 'true' or '$(OutDirFullPath.StartsWith($(Temp), true, null))' == 'true')"/>
```
Since Tmp/Temp are empty, the checks returns true. Figuring where tmp is being picked up.

It turns out that in `env_windows.cmd` we have a `SETLOCAL` but have no `ENDLOCAL`, so the variables end up being local to the script. Adding an ENDLOCAL to the script before defining TMP/TEMP did the trick:
```
[2021-11-22T18:10:49.239Z]   IntDirFullPath=D:\workspace\o3de\build\windows_vs2019\Code\Framework\AzCore\AzCore.dir\profile\, OutDirFullPath=D:\workspace\o3de\build\windows_vs2019\lib\profile\, Tmp=C:\Users\lybuilder\AppData\Roaming\Local\Temp, Temp=C:\Users\lybuilder\AppData\Roaming\Local\Temp
```

Closes Issue #5818

### CODE REVIEW

**Review by amzn-changml — APPROVED** [review]

File: scripts/build/Platform/Windows/build_config.json [hunk 1]
```diff
@@ -134,7 +134,7 @@
       "CMAKE_OPTIONS": "-G \"Visual Studio 16 2019\" -DCMAKE_SYSTEM_VERSION=10.0 -DLY_TEST_IMPACT_INSTRUMENTATION_BIN=%TEST_IMPACT_WIN_BINARY%",
       "CMAKE_LY_PROJECTS": "AutomatedTesting",
       "CMAKE_TARGET": "ALL_BUILD",
-      "CMAKE_NATIVE_BUILD_ARGS": "/m /nologo"
+      "CMAKE_NATIVE_BUILD_ARGS": "/m /nologo /target:DisplayMessage"
```
Comment (lumberyard-employee-dm) [inline scripts/build/Platform/Windows/build_config.json:137]: Will this execute the other msbuild targets or only the DisplayMessages target that was added?

Comment (amznestebanpapp) [reply scripts/build/Platform/Windows/build_config.json:137]: just a temp commit, will take it back, I want to see what Jenkins has for certain MSBuild variables

File: scripts/build/Platform/Windows/env_windows.cmd [hunk 2]
```diff
@@ -22,21 +21,21 @@ IF NOT "%COMMAND_CWD%"=="" (
...
     )
 )
-IF NOT EXIST "!TMP!" (
-    MKDIR "!TMP!"
-)
 
 EXIT /b 0
 
 :error
+ENDLOCAL
```
Comment (amzn-changml) [inline scripts/build/Platform/Windows/env_windows.cmd:40]: Maybe we should put a comment here to explain what happens when we don't do this.

**Review by amzn-changml — APPROVED** [review]

File: scripts/build/Platform/Windows/env_windows.cmd (hunk 2 above)
Comment (amznestebanpapp) [reply scripts/build/Platform/Windows/env_windows.cmd:40]: sure

**Review by lumberyard-employee-dm — APPROVED** [review]

File: scripts/build/Platform/Windows/installer_windows.cmd [hunk 3]
```diff
@@ -17,14 +17,6 @@ IF NOT EXIST %OUTPUT_DIRECTORY% (
 )
 PUSHD %OUTPUT_DIRECTORY%
 
-REM Override the temporary directory used by wix to the workspace (if we have a WORKSPACE_TMP)
-IF NOT "%WORKSPACE_TMP%"=="" (
-    SET "WIX_TEMP=!WORKSPACE_TMP!/wix"
-    IF NOT EXIST "!WIX_TEMP!" (
-        MKDIR "!WIX_TEMP!"
-    )
-)
```
Comment (AMZN-ScottR) [inline scripts/build/Platform/Windows/installer_windows.cmd:26]: If you haven't already, might want to run the installer job on your branch just to confirm this change doesn't mess with Jenkins.

Comment (amznestebanpapp) [reply scripts/build/Platform/Windows/installer_windows.cmd:26]: Yes, run a branch build of the installer and validated that it works

**Review by shirangj — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**